runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageDecoder.java

```diff
@@ -19,6 +20,13 @@
     @Override
     protected void decode(ChannelHandlerContext channelHandlerContext, ByteBuf byteBuf,
                           List<Object> list) throws Exception {
+        // Leave message unchanged if is non-legacy
+        if(byteBuf.getByte(byteBuf.readerIndex()) != API.LEGACY_CORFU_MSG_MARK) {
+            list.add(byteBuf);
+            return;
+        }
+
+        byteBuf.readByte();
```

Comment:
This is a bug. The marker is already consumed in line 24, if that if-statement isnt true, then another byte will be consumed, which is not the marker byte, so this will fail the subsequent call to serialization since it will be missing a byte.